src/azure-cli/azure/cli/command_modules/acr/task.py

```diff
@@ -383,6 +388,8 @@ def acr_task_update(cmd,  # pylint: disable=too-many-locals
         source_triggers = task.trigger.source_triggers
         base_image_trigger = task.trigger.base_image_trigger
         if (commit_trigger_enabled or pull_request_trigger_enabled) or source_triggers:
+            if not branch:
+                branch = _get_branch_name(context_path)
```

Comment:
>_get_branch_name [](start = 25, length = 16)

I notice it's for update scenario, but since the change is migrating to context_path for branch, if `context_path` is `[https://github.com/user/repo.git`](https://github.com/user/repo.git`), should the code set branch to `master`?

Reply:
I do have the same question. Right now the default for update is none unless specified so I kept the same logic while deprecating the change, so we don't introduce a breaking change to the user.